build.gradle

```diff
@@ -59,6 +59,15 @@ subprojects {
 
   configurations {
     testCompile.extendsFrom compileOnly
+
+    compileClasspath {
```

Comment:
This is better than enforcing the exclusion via Checkstyle as this will now be evident at compile time rather than when building the project. Nice.